### ISSUE TITLE
-DPHYLANX_WITH_CXX17 is now required

### ISSUE THREAD

**Issue by stevenrbrandt** [STEllAR-GROUP/phylanx#1032]
When I compile with Phylanx I need to specify the -DPHYLANX_WITH_CXX17 to get things to build correctly. If I don't, I see the error message below:
```

[ 13%] Building CXX object src/CMakeFiles/phylanx_component.dir/execution_tree/primitives/node_data_helpers1d.cpp.o
[ 13%] Building CXX object src/CMakeFiles/phylanx_component.dir/execution_tree/primitives/node_data_helpers3d.cpp.o
In file included from /usr/local/include/hpx/util/runtime_configuration.hpp:11,
                 from /usr/local/include/hpx/runtime.hpp:25,
                 from /usr/local/include/hpx/runtime/parcelset/put_parcel.hpp:10,
                 from /usr/local/include/hpx/runtime/applier/apply.hpp:21,
                 from /usr/local/include/hpx/lcos/packaged_action.hpp:13,
                 from /usr/local/include/hpx/lcos/detail/async_implementations.hpp:13,
                 from /usr/local/include/hpx/lcos/async.hpp:12,
                 from /usr/local/include/hpx/async.hpp:11,
                 from /usr/local/include/hpx/parallel/util/detail/handle_local_exceptions.hpp:11,
                 from /usr/local/include/hpx/parallel/util/partitioner.hpp:24,
                 from /usr/local/include/hpx/parallel/algorithms/for_loop.hpp:34,
                 from /usr/local/include/hpx/include/parallel_for_loop.hpp:9,
                 from /usr/local/include/blaze/math/smp/hpx/DenseVector.h:43,
                 from /usr/local/include/blaze/math/smp/DenseVector.h:46,
                 from /usr/local/include/blaze/math/Band.h:48,
                 from /usr/local/include/blaze/Math.h:46,
                 from /home/jovyan/phylanx/phylanx/ir/node_data.hpp:26,
                 from /home/jovyan/phylanx/phylanx/ast/node.hpp:12,
                 from /home/jovyan/phylanx/phylanx/ast/detail/is_placeholder_ellipses.hpp:10,
                 from /home/jovyan/phylanx/src/ast/detail/is_placeholder_ellipses.cpp:7:
/usr/local/include/hpx/filesystem.hpp:23:26: error: 'filesystem' is not a namespace-name
   23 |     using namespace std::filesystem;
      |                          ^~~~~~~~~~
/usr/local/include/hpx/filesystem.hpp:25:16: error: 'std::filesystem' has not been declared
   25 |     using std::filesystem::canonical;

```

**Comment by hkaiser** [comment]
Since https://github.com/STEllAR-GROUP/hpx/pull/4041 HPX passes the C++ standard used to compile it onto dependent projects. This should be integrated with our build system to avoid having to explicitly specify `PHYLANX_WITH_CXX17`.

**Comment by stevenrbrandt** [comment]
@hkaiser @parsa I notice that if I leave off the -DPHYLANX_WITH_CXX17=1, I get
```
PHYLANX_WITH_CXX:INTERNAL=1
```
in the `CMakeCache.txt`. If I put it on explicitly, I get
```
PHYLANX_WITH_CXX:UNINITIALIZED=1
```
Editing the `CMakeCache.txt` to change `INTERNAL` to `UNINITIALIZED` causes the build to succeed.

**Comment by hkaiser** [comment]
@stevenrbrandt: I'm not sure if anybody is supposed to edit the CMakeCache.txt...

**Comment by stevenrbrandt** [comment]
@hkaiser Of course one isn't supposed to. The point of that test was to verify that changing INTERNAL to UNINITIALIZED was really the key difference between the two configuration methods.

**Comment by stevenrbrandt** [comment]
@hkaiser @parsa https://github.com/STEllAR-GROUP/phylanx/pull/1040

**Comment by hkaiser** [comment]
This was fixed by #1040.
